src/main/java/games/strategy/util/memento/MementoImporter.java

```diff
@@ -0,0 +1,19 @@
+package games.strategy.util.memento;
+
+/**
+ * Imports a memento to its originator.
+ *
+ * @param <T> The type of the memento originator.
+ */
+public interface MementoImporter<T> {
+  /**
+   * Imports the specified memento and returns the originator.
+   *
+   * @param memento The memento to be imported; must not be {@code null}.
+   *
+   * @return The memento originator; never {@code null}.
+   *
+   * @throws MementoImportException If an error occurs while importing the memento.
+   */
+  T importMemento(Memento memento) throws MementoImportException;
```

Comment:
Presumably.

The only specific example of recovery behavior I can cite is what the TripleA client does today when failing to load a save game (the code path in which this method will be called).  An exception thrown by this method will ultimately be handled (probably wrapped a few times) in `GameSelectorModel`, where the recovery action is simply to inform the user of the failure via a message box and then allowing them to select a different save game.